modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -4759,6 +4763,8 @@ private boolean clearLocally0(K key, boolean readers) {
 
         ctx.checkSecurity(SecurityPermission.CACHE_REMOVE);
 
+        ctx.shared().cache().checkReadOnlyState("clear", ctx.config());
```

Comment:
I suppose no neet to check mvcc and security permissions in already readonly cluster ? may be we need to move it upper ?